core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -95,7 +100,7 @@ public function isInitialized(): bool
      *
      * @throws \LogicException
      */
-    public function initialize(): void
+    public function initialize(bool $isFrontend = false): void
```

Comment:
This parameter is not part of the interface. So using it would mean coding against a concrete implementation and not against the contract. The interface should be deprecated and removed later on. Or another solution has to be found. I suggest adding an additional interface for it, maybe with a method named `setMode()` or `initializeFrontend()`.

Reply:
I initially said the same, however, PHP allows the concrete class to have optional parameters that are not part of the method signature in the interface (see https://3v4l.org/KF8DF). So this is fine.

Reply:
Yes, PHP allows it. But then you call the `initialize()` method assuming a concrete implementation. What if another implementation uses `initialize(bool $isBackend = false)`? Isn't it possible to use another impementation? Then there's no sense for an interface. If it is, the interface contract should be used imho.

Reply:
The alternative would be to deprecate the interface. Or use `func_get_args()`.

Reply:
`func_get_args()` would fail as well if another implementation used `initialize(bool $isBackend = false)`.

Reply:
Deprecate the interface and remove the usage of it all together in the core then. I don't see any other possibility.

Reply:
We can discuss it this afternoon, however, we are talking about a legacy class that is additionally marked as `@internal`, so adding another deprecated `ContaoFrameworkInterface` (yes, we already have one) seems a little overeager to me. No hard feelings.

Reply:
I don't mean replacing it by another new interface. Just deprecating it and move to the concrete class instead. No more interface at all. There's no reason to have a second implementation of that interface. There's always only the one `ContaoFramwork` class. Nobody should use any other, I don't see why.

Reply:
I would give you 100 thumbs-up for this answer if it was possible. 😄

Reply:
Changed in 557c06b4fdac87b85cd9957ed7924e3ea988cf35.